### DISCUSSION TITLE
Reveal... How long until reveal actually works...?

### DISCUSSION THREAD

**Discussion by FUCryptoClub** [hashlips-lab/nft-erc721-collection#395] [Q&A]
GM,

How long do you normally have to wait after you update the contract with the reveal function for the metadada to be updated on opensea? (or any other market platform).

I have deployed my contract on rinkeby and everything seems to work fine in terms of contract updates (open and close for whitelist, presale, public sale etc) except the reveal function.

I have attempted to refresh (using refresh button on opensea) the NFTs individually, but it remains as the hidden metadata.

Also on rinkeby.etherscan, if you click into a minted token from the contract, you are faced with a never-ending loading wheel.

I would just like to know if it just takes a little time or there is something wrong.

The set reveal was 9 hours ago.

Look forward to some input.

Thanks again.

**Comment by FUCryptoClub** [comment]
GM,

It's been 1 day 9 hrs since the set reveal, is this normal? Its the only thing stopping me deploying to mainnet now.

Thanks again

**Reply by FUCryptoClub** [reply]
GM @liarco,

Hope you are well.

To help give further insight into my issue, I attempted to use the 'https://testnets-api.opensea.io/api/' force update and got the following error with the API (see screenshot)
<img width="531" alt="Screenshot 2022-08-27 at 09 08 53" src="https://user-images.githubusercontent.com/110253844/187019515-7275eade-88c0-40b5-8329-3d0d22385022.png">

Would it be fact that my IPFS URL files do not have a suffix?

https://ipfs.io/ipfs/bafybeicku.../4676

should it not have a .json at the end... like so....

https://ipfs.io/ipfs/bafybeicku.../4676.json

as I get this error when adding the .json in the browser URL.

ipfs resolve -r /ipfs/bafybeickukd.../4676.json: no link named "4676.json" under bafybeickukd...

Any help would be greatly appreciated.

**Reply by FUCryptoClub** [reply]
Although the hidden metadata works, is this because it is a single file and not stored in a folder? So its not looking specifically for a .json file within a folder unlike for the reveal metadata...

**Reply by FUCryptoClub** [reply]
Looking into it further, I change the following (inside the smartcontract)

`   string memory currentBaseURI = _baseURI();
    return bytes(currentBaseURI).length > 0
        ? string(abi.encodePacked(currentBaseURI, _tokenId.toString(), uriSuffix))
        : '';
  }`

to this

`   string memory currentBaseURI = _baseURI();
    return bytes(currentBaseURI).length > 0
        ? string(abi.encodePacked(currentBaseURI, _tokenId.toString()))
        : '';
  }`

removing this

`, uriSuffix`

Or with this mess things up...?

**Reply by FUCryptoClub** [reply]
removing the `, uriSuffix` seems to have worked after deploying and testing a new contract.

I would be ever so grateful @liarco if you could verify that this is safe to do so, just in case... :)

Thanks again for your amazing work.

**Reply by liarco** [reply]
Hi @FUCryptoClub,
without having access to the actual metadata I may be wrong, but it looks like you are 100% correct.

Usually, whenever you upload a folder containing json files, then they appear with the full filename on IPFS (e.g. `1.json`) but the `uriSuffix` feature is there in order to handle exceptions like the lack of file extension (or any other difference like `.JSON` instead of `.json`).

I hope this helps and I wish you the best of luck with your project.

**Reply by FUCryptoClub** [reply]
@liarco thank you ever so much for your reply. Glad to hear that my investigative research came to a logical conclusion. I will be sure to repay you for your invaluable knowledge and assistance and what you do for the community in general.
